docs/content/concepts/assets/external-assets.mdx

```diff
@@ -0,0 +1,192 @@
+
+## Defining External Assets
+
+### A single external asset
+
+The following code declares a single external asset and passes it to <PyObject object="Definitions"/>
+
+```python file=/concepts/assets/external_assets/single_declaration.py
+from dagster import AssetSpec, Definitions, external_asset_from_spec
+
+defs = Definitions(assets=[external_asset_from_spec(AssetSpec("my_asset"))])
+```
+
+### External assets with dependencies
+
+External assets can depend on other external assets. You do this by using the `deps` argument of <PyObject object="AssetSpec" />.
+
+```python file=/concepts/assets/external_assets/external_asset_deps.py
+from dagster import AssetSpec, Definitions, external_assets_from_specs
+
+asset_one = AssetSpec("asset_one")
+asset_two = AssetSpec("asset_two", deps=[asset_one])
+
+defs = Definitions(assets=external_assets_from_specs([asset_one, asset_two]))
+```
+
+## Keeping an external asset's metadata up to date.
+
+Since with external assets Dagster does not control scheduling or materialization, it is up to the user to keep the metadata for an external asset up-to-date. There are a number of mechanisms for doing so:
+
+- REST APIs
+- Sensors
+- Log events in ops
+
+### Using REST API
+
+Dagster OSS exposes a REST endpoint for reporting asset materializations.
+
+Here's an example `curl` command in a shell script.
+```
+curl --request POST \
+  --url https://path/to/instance/report_asset_materialization/{asset_key}\
+  --header 'Content-Type: application/json' \
+  --data '{
+	"metadata" : {
+		"source": "From curl command,
+	}
+}'
+```
+
+Here's an example of invoking this API in Python with `requests`:
+
+```python
+import requests
+
+url = f"https://path/to/instance/report_asset_materialization/{asset_key}"
+payload = { "metadata": { "source": "From python script" } }
+headers = { "Content-Type": "application/json" }
+
+response = requests.request("POST", url, json=payload, headers=headers)
+```
+
+There are also end points for reporting asset observations and asset check evaluations.
+
+### Using Sensors
+
+You can generate events to attach to external assets and provide them to sensors directly, by using the `asset_events` param of \<PyObject object="SensorResult/>.
+
+```python file=/concepts/assets/external_assets/external_asset_using_sensor.py
+import datetime
+
+from dagster import (
+    AssetMaterialization,
+    AssetSpec,
+    Definitions,
+    SensorResult,
+    external_asset_from_spec,
+    sensor,
+)
+from dagster._core.definitions.sensor_definition import SensorEvaluationContext
+
+
+def utc_now_str() -> str:
+    return datetime.datetime.now(datetime.timezone.utc).strftime("%Y-%m-%d, %H:%M:%S")
+
+
+@sensor()
+def keep_external_asset_a_up_to_date(context: SensorEvaluationContext) -> SensorResult:
+    # materialization happened in external system, but is recorded here
```

Comment:
Should there be some kind of mock check to see if "external_asset_a" has been updated before logging the event?

Reply:
Yeah I guess there is a balance here between staying super minimal versus being "realistic" I generally prefer in-docs examples that are super minimal and use checked in examples for more realism.

Reply:
yeah I find examples with sensors are particularly hard to find that balance. I think this one is probably ok to keep more generic, but i do still thing the first examples in the top part of the page would benefit from being a bit more realistic

Reply:
one thing on this example though, is that the sensor name "keep_external_asset_a_up_to_date" could imply that the sensor is actually updating the data asset in some way. maybe `update_external_asset_a_metadata`?

Reply:
also, is that pattern we're going to recommend that users have one sensor per external asset if they want to update metadata?

Reply:
> also, is that pattern we're going to recommend that users have one sensor per external asset if they want to update metadata?

No I would imagine that most of the time sensors will be used to track sets of assets or all the exhaust from a particular external system.

Reply:
> one thing on this example though, is that the sensor name "keep_external_asset_a_up_to_date" could imply that the sensor is actually updating the data asset in some way. maybe update_external_asset_a_metadata?

Hmmm I guess this depends on how we frame the term "external asset". Does that refer to the represent of the external asset within Dagster, or the physical extenral asset? 

In my mind it was the representation within the Dagster, hence the name `keep_external_asset_a_up_to_date`. Curious what others think. I don't have super strong conviction here so very open to persuasion.

Reply:
> No I would imagine that most of the time sensors will be used to track sets of assets or all the exhaust from a particular external system.

mentioning that as a best practice would be nice. That seems like a question users will have pretty quickly 

> Hmmm I guess this depends on how we frame the term "external asset". Does that refer to the represent of the external asset within Dagster, or the physical external asset?

My initial interpretation was that it was the physical data asset, so definitely interested to see what other people think